alf/utils/process_coordinator_test.py

```diff
@@ -0,0 +1,432 @@
+        coord.join(processes)
+
+    def testClearStopClearsExceptionToo(self):
+        coord = coordinator.Coordinator()
+        ev_1 = Event()
+        processes = [
+            Process(
+                target=raise_on_event,
+                args=(coord, ev_1, None, RuntimeError("First"), True)),
+        ]
+        for t in processes:
+            t.start()
+
+        # not raising with self.assertRaisesRegex(RuntimeError, "First"):
+        ev_1.set()
+        coord.join(processes)
+
+        coord.clear_stop()
+        processes = [
+            Process(
+                target=raise_on_event,
+                args=(coord, ev_1, None, RuntimeError("Second"), True)),
+        ]
+        for t in processes:
+            t.start()
+        # not raising with self.assertRaisesRegex(RuntimeError, "Second"):
+        ev_1.set()
+        coord.join(processes)
+
+    def testRequestStopRaisesIfJoined(self):
+        coord = coordinator.Coordinator()
+        # Join the coordinator right away.
+        coord.join([])
+        reported = False
+        with self.assertRaisesRegex(RuntimeError, "Too late"):
+            try:
+                raise RuntimeError("Too late")
+            except RuntimeError as e:
+                reported = True
+                coord.request_stop(e)
+        self.assertTrue(reported)
+        # If we clear_stop the exceptions are handled normally.
+        coord.clear_stop()
+        try:
+            raise RuntimeError("After clear")
+        except RuntimeError as e:
+            coord.request_stop(e)
+        with self.assertRaisesRegex(RuntimeError, "After clear"):
+            coord.join([])
+
+    def testRequestStopRaisesIfJoined_ExcInfo(self):
+        # Same as testRequestStopRaisesIfJoined but using syc.exc_info().
+        coord = coordinator.Coordinator()
+        # Join the coordinator right away.
+        coord.join([])
+        reported = False
+        with self.assertRaisesRegex(RuntimeError, "Too late"):
+            try:
+                raise RuntimeError("Too late")
+            except RuntimeError:
+                reported = True
+                coord.request_stop(sys.exc_info())
+        self.assertTrue(reported)
+        # If we clear_stop the exceptions are handled normally.
+        coord.clear_stop()
+        try:
+            raise RuntimeError("After clear")
+        except RuntimeError:
+            coord.request_stop(sys.exc_info())
+        with self.assertRaisesRegex(RuntimeError, "After clear"):
+            coord.join([])
+
+
+def _stop_at_0(coord, n, m=None):
+    if n.value == 0:
+        coord.request_stop()
+    else:
+        if m:
+            m.decrement()
+        n.value -= 1
+
+
+class ProcessTest(test.TestCase):
+    def testTargetArgs(self):
+        n = Value(ctypes.c_int, 3)
+        coord = coordinator.Coordinator()
+        p = coordinator.Process.process(
+            coord, target=_stop_at_0, args=(coord, n))
+        coord.join([p])
```

Comment:
coord already registers this process when initializing the Process object, why do you still need to provide it as the argument for join()? It would be convenient if we can just join without any argument.